[#24] Fixed core iterrator and affected modifiers

deluxetom opened the pull request:
* resolves #24
* resolves #23

olivervogel commented:
The number of frames in your test was correct, but strangely the result was no longer animated after encoding to GIF. I have added a check for this.

deluxetom commented:
> The number of frames in your test was correct, but strangely the result was no longer animated after encoding to GIF. I have added a check for this.

Thanks for adding the test, not sure why it's not animated when saving. I'm gonna test locally to see what's going on

deluxetom commented:
@olivervogel while fixing the animation issue, I noticed the loops were lost during the modification process as well. I fixed that part and added a test as well

olivervogel commented:
@deluxetom Sounds good. I haven't been idle either and have added a few more modifiers and analysers.

deluxetom commented:
@olivervogel ok awesome, I'm looking at shapes right now, which modifiers are you working on?
